### PR TITLE
Codeconversion

### PR DESCRIPTION
**Pull request by Raphaeljunior**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Error reporting
```
Comment (jridgewell) [inline README.md]: This file creates a merge conflict. You'll need to rebase off master to fix.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "error-tracker",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "start": "node ./bin/www",
+    "deploy": "gcloud app deploy",
```
Comment (jridgewell) [inline package.json]: And `pretest`, `system-test`, and `e2e-test`? Are they used?

File: routes/error-tracker.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2017 The AMP Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ **/
```
Comment (jridgewell) [inline routes/error-tracker.js]: One too many `*`s.

File: routes/error-tracker.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+const appEngineProjectId = '';
+const logName = 'javascript.errors';
+const SERVER_START_TIME = Date.now();
+const filteredMessageOrException = ['stop_youtube',
+  'null%20is%20not%20an%20object%20(evaluating%20%27elt.parentNode%27)'];
+/**
+ * ERROR_LEVELS
+ * @enum {string}
+ */
+const ERROR_LEVELS = {
```
Comment (jridgewell) [inline routes/error-tracker.js]: We should be able to remove this enum now.

File: routes/error-tracker.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+  }
+
+  let severity = SEVERITY.INFO;
+  /**
+   *if request comes from the cache and thus only from valid
+   *AMP docs we log as "Error"
+   **/
+  let isCdn = false;
+  if (referer.startsWith('https://cdn.ampproject.org/') ||
+    referer.includes('.cdn.ampproject.org/') ||
```
Comment (jridgewell) [inline routes/error-tracker.js]: This should be indented an extra two spaces.

File: routes/error-tracker.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  const referer = params.r;
+  let errorType = 'default';
+  let isUserError = false;
+  if (params.a === '1') {
+    errorType = 'assert';
+    isUserError = true;
+  }
+
+  let severity = SEVERITY.INFO;
+  /**
```
Comment (jridgewell) [inline routes/error-tracker.js]: This comment style is reserved for JSDoc comments only. Inline comments like this should use `//`.

File: routes/error-tracker.js [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+  }
+
+  if (isUserError) {
+    throttleRate = throttleRate / 10;
+  }
+
+  if (sample > throttleRate) {
+    res.set('Content-Type', 'text/plain; charset=utf-8');
+    res.status(statusCodes.OK)
+      .send('THROTTLED\n')
```
Comment (jridgewell) [inline routes/error-tracker.js]: Extra 2 spaces.

File: routes/error-tracker.js [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+
+  if (isFilteredMessageOrException(params.m, exception)) {
+    res.set('Content-Type', 'text/plain; charset=utf-8');
+    res.status(statusCodes.BAD_REQUEST).send('IGNORE\n').end();
+    return;
+  }
+
+  // Don't log testing traffic in production
+  if (params.v.includes('$internalRuntimeVersion$')) {
+    res.sendStatus(statusCodes.NO_CONTENT).end();
```
Comment (jridgewell) [inline routes/error-tracker.js]: Why use `sendStatus`?

File: routes/error-tracker.js [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+    return;
+  }
+
+  // get authentication context for logging
+  let loggingClient = logging({
+    projectId: appEngineProjectId,
+  });
+  let log = loggingClient.log(logName);
+  const resource = {
```
Comment (jridgewell) [inline routes/error-tracker.js]: This can be defined in the `metaData` object.

File: routes/error-tracker.js [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+    res.set('Content-Type', 'application/json; charset=ISO-8859-1');
+    res.status(statusCodes.OK).send(
+      JSON.stringify({
+        message: 'OK\n',
+        event: event,
+        throttleRate: throttleRate,
+      }));
+  } else {
+    res.sendStatus(statusCodes.NO_CONTENT);
+  }
```
Comment (jridgewell) [inline routes/error-tracker.js]: Won't we need to `#end` here?

File: routes/error-tracker.js (hunk 8 above)
Comment (Raphaeljunior) [inline routes/error-tracker.js]: It set's status and sends it back in one go.

File: routes/error-tracker.js (hunk 4 above)
Comment (Raphaeljunior) [inline routes/error-tracker.js]: Its still needed for application logs

**Pull request closed without merging**